frontend/app/(chatroom)/chatroom/[id].tsx

```diff
@@ -0,0 +1,366 @@
+import React, { useEffect, useState } from 'react';
+import { View, Text, Image, StyleSheet, TextInput, TouchableOpacity, FlatList, Alert } from 'react-native';
+import { router, useLocalSearchParams } from 'expo-router';
+import Ionicons from '@expo/vector-icons/Ionicons';
+import { arrayRemove, arrayUnion, deleteDoc, doc, GeoPoint, getDoc, onSnapshot, Timestamp, updateDoc } from 'firebase/firestore';
+import { db } from '@/firebaseConfig';
+import { Chat, Message, Post, User } from '@/app/interfaces';
+import { useUser } from '@/app/UserContext';
+import * as geolib from 'geolib';
+
+interface FoodPost {
+  title: string;
+  image: string;
+  distance: string;
+}
+interface ChatBubble {
+  id: number;
+  text: string;
+  sender: string;
+  timestamp: string;
+}
+export default function ChatRoom() {
+  const [name, setName] = useState<string>('Loading...');
+  const [bubbles, setBubbles] = useState<ChatBubble[]>([]);
+  const [foodPost, setFoodPost] = useState<FoodPost>();
+  const [newMessage, setNewMessage] = useState<string>('');
+  const { user } = useUser();
+  const { id } = useLocalSearchParams();
+
+  /** Calculates the distance between a given post's GeoPoint and the current user's GeoPoint.
+     * Returns a promise of the number (distance) in miles. */
+  const calculateDistanceFromPostToUser = async (postGeoPoint: GeoPoint) => {
```

Comment:
Might have been a better idea to change this duplicated line fragment from home.tsx to be exported here instead of copying the method...

Reply:
yes you're right, but I wanted to prototype quickly.